addons/l10n_in_ewaybill_stock/views/l10n_in_ewaybill_views.xml

```diff
@@ -8,9 +8,10 @@
                 <header>
                     <button name="generate_ewaybill" string="Generate e-Waybill" class="oe_highlight" type="object" invisible="state != 'pending'" data-hotkey="g"/>
                     <button name="cancel_ewaybill" string="Cancel e-Waybill" type="object" invisible="state != 'generated'" data-hotkey="c"/>
-                    <button name="reset_to_pending" string="Reset to Pending" type="object" invisible="state != 'cancel'" data-hotkey="r" confirm="Are you sure you want to re-send the e-Waybill?"/>
+                    <button name="reset_to_pending" string="Reset to Pending" type="object" invisible="not state in ['cancel', 'challan']" data-hotkey="r" confirm="Are you sure you want to reset the e-Waybill / Delivery Challan?"/>
+                    <button name="set_to_challan" string="Use as Challan" type="object" invisible="state != 'pending'" data-hotkey="d"/>
                     <button name="action_export_json" string="Download JSON" class="oe_highlight" type="object" invisible="not error_message" groups="base.group_no_one"/>
-                    <field name="state" widget="statusbar" statusbar_visible="pending,generated"/>
+                    <field name="state" widget="statusbar" statusbar_visible="pending,generated,challan"/>
```

Comment:
```suggestion
                    <field name="state" widget="statusbar" statusbar_visible="pending,generated" invisible="state=='challan'"/>
```
This seems like a nice solution usability-wise to make clear that there is no state transition anymore when you put it as Challan. 
![image](https://github.com/user-attachments/assets/f2911314-fabc-418a-b75b-f908a5265631)

Reply:
```suggestion
                    <field name="state" widget="statusbar" statusbar_visible="pending,generated,challan"  invisible="state=='challan'"/>
```
@jco-odoo You don't want to show the challan state in even in Pending? 👀  (I hope it's just typo)

Reply:
Indeed, I don't want to show it.  The challan is like "not doing the flow of the e-waybill"